tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -157,11 +158,18 @@ func constructConfig() *config.Config {
 	}
 
 	// Detect whether IPv6 is enabled by checking if the pod's IP address is IPv4 or IPv6.
-	podIP, err := getLocalIP()
+	podIPs, err := getLocalIP()
 	if err != nil {
 		panic(err)
 	}
-	cfg.EnableInboundIPv6 = podIP.To4() == nil
+	for _, podIP := range podIPs {
+		// need to check that a proxy can have an IPv6 address but configuration is not configured K8s for dual-stack support.
+		// In this case an ipv6 link local address will appear, but not one that is routable to with K8s
+		if podIP.To4() == nil && podIP.To16() != nil && !podIP.IsLinkLocalUnicast() {
```

Comment:
This is done so much in the code base that maybe it should be moved to its own function.

Reply:
okay, good idea. done